src/lib.rs

```diff
@@ -854,6 +876,20 @@ impl<T, const N: usize> FromFn<T> for [T; N] {
     {
         core::array::from_fn(cb)
     }
+
+    fn try_from_fn<E, F>(mut cb: F) -> Result<Self, E>
+    where
+        F: FnMut(usize) -> Result<T, E>,
+    {
+        // SAFETY: every element of the array below is initialized before use.
+        let mut array: Self = unsafe { mem::zeroed() };
+
+        for i in 0..N {
+            array[i] = cb(i)?;
+        }
+
+        Ok(array)
```

Comment:
I tried to use the pattern described here: https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#initializing-an-array-element-by-element

However I wasn't able to get the final conversion to work, i.e. converting from `[MaybeUninit<T>; N]` to `[T; N]`.

Reply:
`transmute`?

Reply:
Unfortunately `transmute` does not seem to work with const generic arrays:

```
error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
   --> src/lib.rs:891:21
    |
891 |         Ok(unsafe { mem::transmute::<_, Self>(array) })
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: source type: `[MaybeUninit<T>; N]` (this type does not have a fixed size)
    = note: target type: `[T; N]` (this type does not have a fixed size)
```

Reply:
FWIW `core` has a special (unstable) function just for this purpose: https://github.com/rust-lang/rust/blob/d73bd3fb3ba312f3e6b5af4d56d1161d37b71620/library/core/src/mem/maybe_uninit.rs#L947-L957

Reply:
How about casting pointer and using `ptr::read`?

Reply:
Yeah I think a pointer cast should work